packages/taco/src/conditions/base/contract.ts

```diff
@@ -98,9 +143,15 @@ export const contractConditionSchema = rpcConditionSchema
   );
 
 export type ContractConditionProps = z.infer<typeof contractConditionSchema>;
+interface ContractConditionHumanReadableAbi extends ContractConditionProps {
+  functionAbi: string;
+}
 
 export class ContractCondition extends Condition {
-  constructor(value: OmitConditionType<ContractConditionProps>) {
+  constructor(value: OmitConditionType<ContractConditionHumanReadableAbi | ContractConditionProps>) {
+    if (typeof value.functionAbi === 'string') {
+      value.functionAbi = toJsonAbiFormat(value.functionAbi);
+    }
```

Comment:
Why do we need to update this constructor and add a type invariant? Can we update `ContractConditionProps` (`z.infer`) instead?